### PR TITLE
Exports types for ESM module usage

### PR DESCRIPTION
**Pull request by AnotherHermit**
Added types to package json exports so that ESM projects get correct types

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,7 +26,8 @@
     ".": {
       "require": "./dist/index.js",
       "import": "./dist/react-use-intercom.esm.js",
-      "browser": "./dist/react-use-intercom.esm.js"
+      "browser": "./dist/react-use-intercom.esm.js",
+      "types": "./dist/index.d.ts"
```
Comment (devrnt) [inline package.json:30]: Can you move this to the first line, according to https://www.typescriptlang.org/docs/handbook/esm-node.html#packagejson-exports-imports-and-self-referencing the `types` entry must occur first

**Review by devrnt — CHANGES_REQUESTED** [review]
Thanks for this PR!

File: package.json (hunk 1 above)
Comment (AnotherHermit) [reply package.json]: Of course! It has been updated now

Comment (devrnt) [reply package.json]: Thanks

**Review by devrnt — APPROVED** [review]

**Comment by devrnt** [comment]
[v3.0.1](https://github.com/devrnt/react-use-intercom/releases/tag/v3.0.1)

**Pull request merged**